### ISSUE TITLE
[Feature]: add example with usage of both `keyof` & `typeof` operators

### ISSUE THREAD

**Issue by reznikovAndrey** [typehero/typehero#1307]
### Description

I think it's necessary to provide example with usage of both `keyof` & `typeof` operator.
The final solution like this will be more clear.
<img width="924" alt="Снимок экрана 2023-12-11 в 01 46 49" src="https://github.com/typehero/typehero/assets/81411663/024cf006-b1ed-441c-ab71-37bea59583e3">
 

### Explanation

In my opinion, after completing [challenge with usage of `keyof` operator](https://typehero.dev/challenge/keyof) newcomer might think, that provided examples are only valid way of using `keyof`.

In other words:
1. You create new type with typeof.
2. You need use `keyof` with this type.

But in real world there are a lot of examples of using both `keyof` & `typeof` operators together.
I consider this as best practice.

**Comment by bautistaaa** [comment]
these are meant to be taken in order when you enroll in the foundations track. perhaps we should remove these from the explore page and point them to the track

**Comment by bautistaaa** [comment]
i also don’t understand the best practice part of your comment. the pref would usually be typeof standalone so you can reuse the type and making a separate type if you needed the union

**Comment by reznikovAndrey** [comment]
Yeah, you are right, that's situational.
My fault, it's not best practice.

At least may be it would be better to have explicit example of usage of this operators together?

<img width="498" alt="Снимок экрана 2023-12-11 в 14 35 59" src="https://github.com/typehero/typehero/assets/81411663/1c57dcea-2e77-4d76-9783-f376b551b885">
